sandbox/src/components/renderingZone.tsx

```diff
@@ -176,8 +213,8 @@ export class RenderingZone extends React.Component<IRenderingZoneProps> {
             }
         } else {
             var pbrPresent = false;
-            for (var i = 0; i < this._scene.materials.length; i++) {
-                if (this._scene.materials[i].transparencyMode !== undefined) {
+            for (const material of this._scene.materials) {
+                if (material instanceof PBRBaseMaterial) {

```

Comment:
instanceof is relatively expensive (and sometimes unreliable). Is there a different way of checking its type? will `getClassName()` work here?

Reply:
I'm sure there is. I don't like it because that is what caused the bug to begin with. We checked for transparencyMode on PBR, but later we added transparencyMode to StandardMaterial.

`getClassName` can work, but we would have to check a few strings since we want all materials inherited from `PBRBaseMaterial`.

I'm not too worried about the expensive part since this only runs once. How is it unreliable?

Reply:
it checks prototypes. A class with the same prototypes will return true. Rare but possible :-)

Reply:
Yes, I agree, but if it has the same prototype, it's probably okay?